src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/delete_interaction_choice_set_request.cc

```diff
@@ -87,23 +89,46 @@ void DeleteInteractionChoiceSetRequest::Run() {
     return;
   }
   SendVrDeleteCommand(app);
+}
 
-  smart_objects::SmartObject msg_params =
-      smart_objects::SmartObject(smart_objects::SmartType_Map);
+bool DeleteInteractionChoiceSetRequest::Init() {
+  hash_update_mode_ = HashUpdateMode::kDoHashUpdate;
+  return true;
+}
 
-  msg_params[strings::interaction_choice_set_id] = choice_set_id;
-  msg_params[strings::app_id] = app->app_id();
+void DeleteInteractionChoiceSetRequest::on_event(
+    const event_engine::Event& event) {
+  using namespace helpers;
+  LOG4CXX_AUTO_TRACE(logger_);
 
-  app->RemoveChoiceSet(choice_set_id);
+  if (event.id() == hmi_apis::FunctionID::VR_DeleteCommand) {
+    {
+      sync_primitives::AutoLock auto_lock(requests_lock_);
```

Comment:
i think it is good practice to lock this lock just immediately before the protected data is accessed, i would move this line immediately above `sent_requests_.find`

Reply:
@iCollin fixed in https://github.com/smartdevicelink/sdl_core/pull/2601/commits/59029ef38e9f4161255d6a6f929ff3bdf7870d68